src/api/middleware/user-data.middleware.js

```diff
@@ -0,0 +1,20 @@
+import {Exception} from "../exceptions/Exception.js";
+
+function validateStringNotEmpty(string, errorMessagePrefix, errors) {
+    if (!string || string.trim().length === 0) {
+        errors.push(`${errorMessagePrefix} is empty`);
+    }
+}
+
+export const checkUserDataMiddleware = (req, res, next) => {
+    const userData = req.body;
+    const errors = [];
+    validateStringNotEmpty(userData.id, "User id", errors);
+    validateStringNotEmpty(userData.name, "User name", errors);
```

Comment:
not a good practice to pass an object as a parameter and mutate it
validateStringNotEmpty can return string or undefined

```js
const errors = [validateStringNotEmpty(userData.id, "User id"), validateStringNotEmpty(userData.name, "User name")].filter(x => x)
```